keras/backend/cntk_backend.py

```diff
@@ -2573,7 +2573,9 @@ def reverse(x, axes):
 
 
 def slice(x, start, size):
-    raise NotImplementedError
+    out = x[tuple([py_slice(i, i + j) for (i, j) in zip(start, size)])]
```

Comment:
This could potentially return without crashing despite incorrect values for `start` or `size` (due to the use of `zip`). I would suggest adding a check that both `start` and `size` are tuples of the same length as `ndim(x)`.

Reply:
@fchollet, The sanity check you proposed has been added.